text/XXXX-browser-traces.md

```diff
@@ -0,0 +1,98 @@
+
+# Motivation
+
+Today there are two ways to start a trace: they can be started on the backend and then
+continued on the client, or they can be started on the client.  In either case however
+client SDKs are likely to create more traces that are disconnected to the backend trace
+on page navigation or view changes.  This creates the situation that today the only sensible
+way for dynamic sampling is to have traces started uniformly on the frontend project as
+otherwise the dynamic sampling rules from both projects need to be modified (head trace
+is where dynamic sampling rules apply).
+
+The secondary motivation is that creating new traces on navigation also wipes out the
+causal relationship to what happened before.  For instance it's more than possible that
+before a client-side navigation the state of the application corrupted, but we lose that
+trace relationship and a user has to manually piece it back together by for instance
+listing all transactions created by a specific user ID.
+
+# Background
+
+We became aware of this problem in two ways recently:
+
+1. Users want to sometimes create a transaction within another transaction.  Today there
+   is no way to link these together for the purpose of dynamic sampling.  A separate RFC
+   [0014](https://github.com/getsentry/rfcs/pull/14) is proposed to add an explicit way
+   to carry forward the sampling context for a new transaction started after an already
+   existing one.  It works by explicitly continuing the trace.  This solves part of this
+   issue, but it leaves out the case where the sampling context naturally moves to another
+   Sentry project.
+
+2. We wanted to change our own tracing integration to start tracing on the server
+   [Sentry PR #39349](https://github.com/getsentry/sentry/pull/39349) where this would
+   require mirroring the sampling settings to another Sentry project and would also affect
+   API requests detatched from user sessions.
+
+# Supporting Data
+
+The [honeycomb whitepaper on front-end observability](https://www.honeycomb.io/wp-content/uploads/2022/03/Front-end-Observability-Whitepaper-1.pdf)
+recommends continung traces from the server until the natural end of the user Session:
+
+> To accomplish this task, you will use the first event (page load) as the start of your trace and
+> connect that first event to additional spans to build a full trace of the user session. Each span
+> will represent a single thing that you want to track, such as a server request or a user click.
+
+# Options Considered
+
+There are multiple ways in which this problem can be addressed.
+
+## Encouraged Root Trace Project and Session Long Traces
+
+In the most trivial case the recommendation to customers would be to pick one project that
+starts traces for real user sessions.  This could be *either* frontend of backend but it
+should attempt to be consistent about it.  In either case the client SDK should *continue the
+trace* until the browser tab naturally closes.
```

Comment:
The issue here is with traces that then can be multiple hours long - and so then the value of analyzing a trace is completely lost.

I'm strongly against this because I feel like it'll reduce the value of trace view, and make it harder for us to expand the tracing product.

Reply:
That sounds a bit like a UX issue? I don't think that it's inherently problematic if a trace is very long, we today already do not draw much of a value out of the trace view.

To me it feels at least like there is more value to the trace being connected than the trace being split into transactions on every navigation.

Reply:
User inactivity in a browser tab could be another natural way of ending a session, and beginning a new one. 

There could be users that leave tabs open for a very long time, and having gaps between transactions might seem weird.